spp_event_data/models/registrant.py

```diff
@@ -0,0 +1,43 @@
+# Part of OpenSPP. See LICENSE file for full copyright and licensing details.
+
+from datetime import date
+
+from odoo import _, fields, models
+
+
+class OpenSPPRegistrant(models.Model):
+    _inherit = "res.partner"
+
+    event_data_ids = fields.One2many("spp.event.data", "registrar", "Event Data IDs")
+
+    def _compute_active_event(self, model):
+        active_event = self.env["spp.event.data"].search(
+            [
+                ("registrar", "=", self.id),
+                ("model", "=", model),
+                ("expiry_date", "=", False),
+            ]
+        )
+        active_event_id = None
+        if active_event:
+            active_event_id = active_event[0].id
+        return active_event_id
+
+    def end_active_event(self, res_id):
```

Comment:
This function should be on [OpenSPPEventData(models.Model):](https://github.com/OpenSPP/openspp-registry/blob/ea7d8f2935c2822fdb97334cbb2c2b756792754a/spp_event_data/models/event_data.py#L10)